API connector: integration fails to start due to missing operation

I have a `DB->Todo API` integration that should create new record via todo-example API, but the Camel context fails to start with the following exception.

The naming of operations seem to be incorrect?
```
Failed to create route flow1: Route(flow1)[[From[sql-start-connector-1?query=SELECT+*+FROM... because of Failed to create Producer for endpoint: swagger-operation-3. Reason: java.lang.IllegalArgumentException: The specified operation with ID: `null` cannot be found in the Swagger specification loaded from `file:/tmp/swagger-operation-3404004334764459470.swagger`. Operations defined in the specification are: operation-0, operation-1, operation-2, operation-3, operation-4
```


<details>
<summary>Exception in log</summary>

```
Error starting ApplicationContext. To display the auto-configuration report re-run your application with 'debug' enabled.
  | 2018-01-10 09:55:56.078 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application startup failed
  |  
  | org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route flow1: Route(flow1)[[From[sql-start-connector-1?query=SELECT+*+FROM... because of Failed to create Producer for endpoint: swagger-operation-3. Reason: java.lang.IllegalArgumentException: The specified operation with ID: `null` cannot be found in the Swagger specification loaded from `file:/tmp/swagger-operation-3404004334764459470.swagger`. Operations defined in the specification are: operation-0, operation-1, operation-2, operation-3, operation-4
  | at org.apache.camel.util.ObjectHelper.wrapRuntimeCamelException(ObjectHelper.java:1831) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:136) ~[camel-spring-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.spring.SpringCamelContext.onApplicationEvent(SpringCamelContext.java:174) ~[camel-spring-2.20.1.jar!/:2.20.1]
  | at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.12.RELEASE.jar!/:4.3.12.RELEASE]
  | at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.8.RELEASE.jar!/:1.5.8.RELEASE]
  | at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
  | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_141]
  | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_141]
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_141]
  | at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_141]
  | at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
  | at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
  | at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
  | at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:587) [project-0.1-SNAPSHOT.jar:na]
  | Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route flow1: Route(flow1)[[From[sql-start-connector-1?query=SELECT+*+FROM... because of Failed to create Producer for endpoint: swagger-operation-3. Reason: java.lang.IllegalArgumentException: The specified operation with ID: `null` cannot be found in the Swagger specification loaded from `file:/tmp/swagger-operation-3404004334764459470.swagger`. Operations defined in the specification are: operation-0, operation-1, operation-2, operation-3, operation-4
  | at org.apache.camel.impl.RouteService.warmUp(RouteService.java:147) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.doWarmUpRoutes(DefaultCamelContext.java:3945) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.safelyStartRouteServices(DefaultCamelContext.java:3852) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.doStartOrResumeRoutes(DefaultCamelContext.java:3638) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.doStartCamel(DefaultCamelContext.java:3490) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.access$000(DefaultCamelContext.java:208) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3249) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3245) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3268) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.doStart(DefaultCamelContext.java:3245) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:3168) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:133) ~[camel-spring-2.20.1.jar!/:2.20.1]
  | ... 24 common frames omitted
  | Caused by: org.apache.camel.FailedToCreateProducerException: Failed to create Producer for endpoint: swagger-operation-3. Reason: java.lang.IllegalArgumentException: The specified operation with ID: `null` cannot be found in the Swagger specification loaded from `file:/tmp/swagger-operation-3404004334764459470.swagger`. Operations defined in the specification are: operation-0, operation-1, operation-2, operation-3, operation-4
  | at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:578) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.ProducerCache.acquireProducer(ProducerCache.java:168) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.processor.SendProcessor.doStart(SendProcessor.java:248) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.processor.RedeliveryErrorHandler.doStart(RedeliveryErrorHandler.java:1472) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:44) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:31) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.processor.interceptor.DefaultChannel.doStart(DefaultChannel.java:160) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:62) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.processor.MulticastProcessor.doStart(MulticastProcessor.java:1172) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.processor.DelegateAsyncProcessor.doStart(DelegateAsyncProcessor.java:80) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.RouteService.startChildService(RouteService.java:370) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.RouteService.doWarmUp(RouteService.java:196) ~[camel-core-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.RouteService.warmUp(RouteService.java:145) ~[camel-core-2.20.1.jar!/:2.20.1]
  | ... 36 common frames omitted
  | Caused by: java.lang.IllegalArgumentException: The specified operation with ID: `null` cannot be found in the Swagger specification loaded from `file:/tmp/swagger-operation-3404004334764459470.swagger`. Operations defined in the specification are: operation-0, operation-1, operation-2, operation-3, operation-4
  | at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.createProducer(RestSwaggerEndpoint.java:182) ~[camel-rest-swagger-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.component.connector.DefaultConnectorEndpoint.createProducer(DefaultConnectorEndpoint.java:51) ~[camel-connector-2.20.1.jar!/:2.20.1]
  | at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:573) ~[camel-core-2.20.1.jar!/:2.20.1]
  | ... 70 common frames omitted
```

</details>

--- a/app/rest/project-generator/src/main/java/io/syndesis/project/converter/visitor/ConnectorStepVisitor.java
+++ b/app/rest/project-generator/src/main/java/io/syndesis/project/converter/visitor/ConnectorStepVisitor.java
@@ -29,6 +29,7 @@ import io.syndesis.dao.manager.DataManager;
 import io.syndesis.model.Split;
 import io.syndesis.model.WithConfigurationProperties;
 import io.syndesis.model.action.ConnectorAction;
+import io.syndesis.model.action.ConnectorDescriptor;
 import io.syndesis.model.connection.Connection;
 import io.syndesis.model.connection.Connector;
 import io.syndesis.model.integration.Step;
@@ -158,6 +159,10 @@ public class ConnectorStepVisitor implements StepVisitor {
                 .filter(Predicates.or(connector::isSecret, action::isSecret))
                 .forEach(e -> e.setValue(String.format("{{%s-%d.%s}}", componentScheme, visitorContext.getIndex(), e.getKey())));
         }
+
+        // any configuredProperties on action descriptor are considered
+        final ConnectorDescriptor descriptor = action.getDescriptor();
+        properties.putAll(descriptor.getConfiguredProperties());
 
         if (hasComponentProperties(configuredProperties, connector, action)) {
             return new io.syndesis.integration.model.steps.Endpoint(String.format("%s-%d", componentScheme, visitorContext.getIndex()), Map.class.cast(properties));

